src/elexmodel/handlers/data/Estimandizer.py

```diff
@@ -45,44 +37,45 @@ def check_and_create_estimands(self, data_df, estimands, historical, current_dat
         if not historical and len(results_column_names) == 0:
             raise EstimandException("This is not a test election, it is missing results data")
 
-        return (data_df, columns_to_return)
+        return data_df, columns_to_return
 
-    def add_estimand_baselines(self, data_df, estimand_baselines, historical):
+    def add_estimand_baselines(self, data_df, estimand_baselines, historical, include_results_estimand=False):
         # if we are in a historical election we are only reading preprocessed data to get
         # the historical election results of the currently reporting units.
         # so we don't care about the total voters or the baseline election.
 
         for estimand, pointer in estimand_baselines.items():
             if pointer is None:
-                # should only happen when we're going to create a new estimand
+                # when we are creating a new estimand
                 pointer = estimand
 
             baseline_col = f"{BASELINE_PREFIX}{pointer}"
 
             if baseline_col not in data_df.columns:
-                # will raise a KeyError if a function with the same name as `pointer` doesn't exist
-                data_df = globals()[pointer](data_df)
-                results_col = f"{RESULTS_PREFIX}{estimand}"
-                data_df[results_col] = data_df[baseline_col].copy()
+                data_df = globals()[estimand](data_df, BASELINE_PREFIX)
 
             if not historical:
-                # Adding one to prevent zero divison
                 data_df[f"last_election_results_{estimand}"] = data_df[baseline_col].copy() + 1
 
+        if include_results_estimand:
+            data_df, ___ = self.check_and_create_estimands(data_df, estimand_baselines.keys(), historical)
+
         return data_df
 
 
 # custom estimands
 
 
-def party_vote_share_dem(data_df):
-    # should only happen when we're replaying an election
-    if f"{BASELINE_PREFIX}dem" not in data_df.columns and f"{BASELINE_PREFIX}turnout" not in data_df.columns:
-        data_df[f"{RESULTS_PREFIX}party_vote_share_dem"] = (
-            data_df[f"{RESULTS_PREFIX}dem"] / data_df[f"{RESULTS_PREFIX}turnout"]
-        )
+def party_vote_share_dem(data_df, col_prefix):
+    if f"{col_prefix}dem" in data_df.columns and f"{col_prefix}turnout" in data_df.columns:
```

Comment:
I was testing this out with the test bed and the `2021-11-02_VA_G --office_id Y --geographic_unit_type "precinct-district" --no_ap` command we've been talking about.  In that data set, there are only "dem" and "total" columns, nothing like `results_dem` or `baseline_dem` like we'd need here.  So I had to add this check and use "dem" and "total" columns as a fall-back 🤔

Reply:
ooh yes, I know what you mean! I think it would be better to deal with that in the testbed. At some point I changed the name of the column that was being stored in s3 and I think the model should assume that whatever we pass to it starts with `results_dem` etc.

Reply:
Yeah, I agree.  As I was working on this I was wondering how often we'd be updating estimandizers to accommodate new column names lol 😅  I'll take this out and add some logic in the test bed.  Thanks 😄